app/lib/service/openid/jwt.dart

```diff
@@ -123,7 +114,179 @@ Map<String, dynamic> _decodePart(String part) {
+        kid = map['kid'] as String?;
+
+  factory JwtHeader(Map<String, dynamic> values) {
+    try {
+      return JwtHeader._(values);
+    } on FormatException {
+      rethrow;
+    } catch (_) {
+      throw FormatException('Unexpected value in JWT header.');
+    }
+  }
+}
+
+/// The parsed JWT payload.
+class JwtPayload extends UnmodifiableMapView<String, dynamic> {
+  /// timestamp when token was issued
+  final DateTime? iat;
+
+  /// timestamp before which the token is not valid
+  final DateTime? nbf;
+
+  /// timestamp at which the token expires
+  final DateTime? exp;
+
+  JwtPayload._(super.map)
+      : iat = _parseSecondsIfNotNull(map['iat'] as int?),
+        nbf = _parseSecondsIfNotNull(map['nbf'] as int?),
+        exp = _parseSecondsIfNotNull(map['exp'] as int?);
+
+  factory JwtPayload(Map<String, dynamic> map) {
+    try {
+      return JwtPayload._(map);
+    } on FormatException {
+      rethrow;
+    } catch (_) {
+      throw FormatException('Unexpected value in JWT payload.');
+    }
+  }
+
+  /// Returns the first key from [keys] which is missing or has `null` value.
+  ///
+  /// Returns `null` if all the [keys] are present with non-null values.
+  String? firstMissingKey(Iterable<String> keys) {
+    for (final key in keys) {
+      final value = this[key];
+      if (value == null) {
+        return key;
+      }
+    }
+    return null;
+  }
+
+  /// Verifies the timestamps with to the provided [now]:
+  /// - [iat] <= [now]
+  /// - [nbf] <= [now]
+  /// - [now] <= [exp]
+  bool verifyTimestamps([DateTime? now]) {
+    now ??= clock.now();
+    if (iat != null && iat!.isAfter(now)) {
+      return false;
+    }
+    if (nbf != null && nbf!.isAfter(now)) {
+      return false;
+    }
+    if (exp != null && exp!.isBefore(now)) {
+      return false;
+    }
+    return true;
+  }
+}
+
+/// Parsed payload with the payload values GitHub sends with the token.
+class GitHubJwtPayload extends JwtPayload {
+  /// timestamp when token was issued
+  @override
+  DateTime get iat => super.iat!;
+
+  /// timestamp before which the token is not valid
+  @override
+  DateTime get nbf => super.nbf!;
+
+  /// timestamp at which the token expires
+  @override
+  DateTime get exp => super.exp!;
+
+  /// user controllable URL identifying the intended audience
+  final String aud;
+
+  /// repository for which the action is running
+  final String repository;
+
+  /// owner of the repository
+  final String repositoryOwner;
+
+  /// name of the event that triggered the workflow
+  final String eventName;
+
+  /// git reference (e.g. branch name or tag name)
+  final String ref;
+
+  /// the kind of git reference given (e.g. "branch")
+  final String refType;
+
+  /// name of the environment used by the job
+  final String environment;
+
+  static const _keys = <String>[
+    'iat',
+    'nbf',
+    'exp',
+    'aud',
+    'repository',
+    'repository_owner',
+    'event_name',
+    'ref',
+    'ref_type',
+    'environment',
+  ];
```

Comment:
Why not use an set? if there is something important to the ordering, then that should be documented.
```suggestion
  static const _requiredClaims = <String>{
    'iat',
    'nbf',
    'exp',
    'aud',
    'repository',
    'repository_owner',
    'event_name',
    'ref',
    'ref_type',
    'environment',
  };
```

Set also allows us to do `_requiredClaims.difference({...payload.keys}).sorted()`, by using set difference operation.